packages/devtools-reps/src/object-inspector/component.js

```diff
@@ -187,33 +178,15 @@ class ObjectInspector extends Component<Props> {
     }
 
     const {
-      createObjectClient,
-      createLongStringClient,
-      loadedProperties,
       nodeExpand,
       nodeCollapse,
       recordTelemetryEvent,
       roots
     } = this.props;
 
     if (expand === true) {
-      const gripItem = getClosestGripNode(item);
-      const value = getValue(gripItem);
-      const isRoot =
-        value &&
-        roots.some(root => {
-          const rootValue = getValue(root);
-          return rootValue && rootValue.actor === value.actor;
-        });
-      const actor = isRoot || !value ? null : value.actor;
-      nodeExpand(
-        item,
-        actor,
-        loadedProperties,
-        createObjectClient,
-        createLongStringClient
-      );
-
+      const actor = getActor(item, roots);
```

Comment:
i feel like this could also be written differently to say something like, fetch every item that has an actor and isn't an environment... 

If we did this, then we would not need to pass actor in at all. And noadLoadProperties would look like this:

```js
function nodeLoadProperties(node: Node, actor) {
  return async ({ dispatch, client, getState }: ThunkArg) => {
    const loadedProperties = getLoadedProperties(getState());
    if (node.actor && !isEnvironment(node.actor) && loadedProperties.has(node.path)) {
      return;
    }
    ...
}
```

The advantage of this approach is 

1. we don't have to do the complicated roots check
2. we can express the logic in the action, which is easier to test and understand